### PR TITLE
Search in RECENT ASSETS FROM MY PROJECTS in Welcome Screen Designer is Not Working Correctly 

### PR DESCRIPTION
**Pull request by marcoAntonioNina**
## Issue & Reproduction Steps
Filtering on the recent assets page is not working correctly.

Expected behavior:
We should be able to filter the listing results.
Actual behavior:
does not perform filtering

## Solution
- Change query and add filtering.

## How to Test
Test the steps in the ticket

## Related Tickets & Packages
- [FOUR-12457](https://processmaker.atlassian.net/browse/FOUR-12457)

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:next
ci:package-projects:bugfix/FOUR-12457
ci:deploy

[FOUR-12457]: https://processmaker.atlassian.net/browse/FOUR-12457?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-7beeae43ca.engk8s.processmaker.net

**Comment by pmPaulis** [comment]
@marcoAntonioNina 
I tested in the build generated and please check the following observations in the functionality:

1. When asset Data connector was added, the server is giving a error 
image

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-7beeae43ca.engk8s.processmaker.net

**Comment by pmPaulis** [comment]
@marcoAntonioNina 
I reviewed with the last changes and this is working as expected 

https://github.com/ProcessMaker/processmaker/assets/42388723/3bae74d7-a3d0-45df-afe0-7c70fe34b297

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-7beeae43ca.engk8s.processmaker.net

**Review by pmPaulis — APPROVED** [review]

**Comment by pmPaulis** [comment]
@marcoAntonioNina 
I can not test the Process List, currently this has some other inconsistencies with OWNER column but this is not related with your changes

File: ProcessMaker/Models/Process.php [hunk 1]
```diff
@@ -1702,8 +1702,10 @@ public function scopeFilter($query, $filterStr)
             $query->where('processes.name', 'like', $filter)
                  ->orWhere('processes.description', 'like', $filter)
                  ->orWhere('processes.status', '=', $filterStr)
-                 ->orWhere('user.firstname', 'like', $filter)
-                 ->orWhere('user.lastname', 'like', $filter)
+                 ->with(['user' => function ($query) use ($filter) {
```
Comment (eiresendez) [inline ProcessMaker/Models/Process.php:1705]: hi @marcoAntonioNina this change causes the search in `/processes`by "user" to stop working. I believe this query would be more suitable for the cases we handle:

```
->orWhereHas('user', function ($query) use ($filter) {
    $query->where('firstname', 'like', $filter)
          ->orWhere('lastname', 'like', $filter);
})
```

**Review by eiresendez — CHANGES_REQUESTED** [review]

File: ProcessMaker/Models/Process.php (hunk 1 above)
Comment (marcoAntonioNina) [reply ProcessMaker/Models/Process.php]: Thanks @eiresendez , the change was updated.

**Review by eiresendez — APPROVED** [review]

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-7beeae43ca.engk8s.processmaker.net

**Comment by pmPaulis** [comment]
@marcoAntonioNina 
The process lis is working fine 

https://github.com/ProcessMaker/processmaker/assets/42388723/cc9e2f9e-7a73-4120-97f3-8b788ce4fbd7

**Pull request merged**